superset-frontend/src/explore/components/ExploreViewContainer/index.jsx

```diff
@@ -768,6 +779,30 @@ function mapStateToProps(state) {
     dashboardId = undefined;
   }
 
+  if (
+    form_data.viz_type === 'big_number_total' &&
+    slice?.form_data?.subheader &&
+    (!controls.subtitle?.value || controls.subtitle.value === '')
+  ) {
+    controls.subtitle = {
+      ...controls.subtitle,
+      value: slice.form_data.subheader,
+    };
+  }
+  if (
+    form_data.viz_type === 'big_number_total' &&
+    slice?.form_data?.subheader_font_size &&
+    (!controls.subtitle_font_size?.value ||
+      controls.subtitle_font_size.value === '')
+  ) {
+    controls.subtitle_font_size = {
+      ...controls.subtitle_font_size,
+      value: slice.form_data.subheader_font_size,
+    };
+  }
```

Comment:
I'm suggesting this change for two reasons:
- It moves the second change (`controls.subtitle_font_size =  slice.form_data.subheader_font_size`) as part of the first `if` block. This is mostly because if the chart has a `subheader` config and does not have a `subtitle` config, then `subheader_font_size` should be the source of truth. For safety, just checking if `slice?.form_data?.subheader_font_size` exists.
- In the current implementation, `!controls.subtitle_font_size?.value` is evaluated. I believe `controls.subtitle_font_size` always gets a default value, so that second `if` would never really execute and instead the old `subheader` ends up with the default size.


```suggestion
  if (
    form_data.viz_type === 'big_number_total' &&
    slice?.form_data?.subheader &&
    (!controls.subtitle?.value || controls.subtitle.value === '')
  ) {
    controls.subtitle = {
      ...controls.subtitle,
      value: slice.form_data.subheader,
    };
    if (slice?.form_data?.subheader_font_size) {
      controls.subtitle_font_size = {
        ...controls.subtitle_font_size,
        value: slice.form_data.subheader_font_size,
      };
    }
  }
```